src/elm/reusable.js

```diff
@@ -50,20 +50,43 @@ class FunctionRef extends Expression {
     this.library = json.libraryName;
   }
   exec(ctx) {
-    let functionDef, child_ctx;
+    let functionDefs, child_ctx;
     if (this.library) {
       const lib = ctx.get(this.library);
-      functionDef = lib ? lib.get(this.name) : undefined;
+      functionDefs = lib ? lib.getFunction(this.name) : undefined;
       const libCtx = ctx.getLibraryContext(this.library);
       child_ctx = libCtx ? libCtx.childContext() : undefined;
     } else {
-      functionDef = ctx.get(this.name);
+      functionDefs = ctx.get(this.name);
       child_ctx = ctx.childContext();
     }
     const args = this.execArgs(ctx);
-    if (args.length !== functionDef.parameters.length) {
-      throw new Error('incorrect number of arguments supplied');
+
+    // Filter out functions w/ wrong number of arguments.
+    functionDefs = functionDefs.filter(f => f.parameters.length === args.length);
+    // If there is still > 1 matching function, filter by argument types
+    if (functionDefs.length > 1) {
+      functionDefs = functionDefs.filter(f => {
+        let match = true;
+        for (let i = 0; i < args.length && match; i++) {
+          if (args[i] !== null) {
+            match = ctx.matchesTypeSpecifier(args[i], f.parameters[i].operandTypeSpecifier);
```

Comment:
NOTE: I don't know if it is necessary, but if we wanted to be extra safe, we could check if `operandTypeSpecifier` is null, and if so, convert `operandType` to a `NamedTypeSpecifier` (similar to [here](https://github.com/cqframework/cql-execution/blob/08bf598ed5093a463eedc90ea6c3baa0fd829b58/src/elm/type.js#L16-L22)) (assuming `operandType` is not also null).

But I _think_ in practice CQL to ELM only ever uses `operandTypeSpecifier`.